diff --git a/build/.moduleignore b/build/.moduleignore
index 3f573e06078..63073e3bc02 100644
--- a/build/.moduleignore
+++ b/build/.moduleignore
@@ -161,8 +161,9 @@ typescript/lib/tsserverlibrary.js
 # We still need to include stdlib d.ts
 !typescript/lib/lib*.d.ts
 
-jschardet/index.js
-jschardet/src/**
+# RiteMark fix: keep jschardet source files for Node.js require() in git extension
+# jschardet/index.js
+# jschardet/src/**
 # TODO@esm uncomment when we can use jschardet.min.js again jschardet/dist/jschardet.js
 
 es6-promise/lib/**
